pydicom/valuerep.py

```diff
@@ -831,19 +865,39 @@ def encode(self, encodings=None):
                 self.components, encodings or [default_encoding])
         return self.original_string
 
-    def family_comma_given(self):
+    def family_comma_given(self) -> str:
+        """Return the name as "Family, Given"."""
         return self.formatted('%(family_name)s, %(given_name)s')
 
-    def formatted(self, format_str):
-        self._create_dict()
-        return format_str % self._dict
+    def formatted(self, format_str: str) -> str:
+        """Return the name as a :class:`str` formatted using `format_str`."""
+        return format_str % self._create_dict()
 
-    def __bool__(self):
+    def __bool__(self) -> bool:
+        """Return ``True`` if the name is not empty."""
         if self.original_string is None:
-            return (bool(self._components) and
-                    (len(self._components) > 1 or bool(self._components[0])))
+            return (
+                bool(self._components)
+                and (
+                    len(self._components) > 1 or bool(self._components[0])
+                )
+            )
+
         return bool(self.original_string)
 
 
 # Alias old class names for backwards compat in user code
-PersonNameUnicode = PersonName = PersonName
+def __getattr__(name):
+    if name == "PersonNameUnicode":
+        warnings.warn(
+            "'PersonNameUnicode' is deprecated and will be removed in "
+            "pydicom v2.2, use 'PersonName' instead",
+            DeprecationWarning
+        )
+        return globals()['PersonName']
+
+    raise AttributeError(f"module {__name__} has no attribute {name}")
+
+
+if sys.version_info[:2] < (3, 7):
+    PersonNameUnicode = PersonName
```

Comment:
why is this needed?

Reply:
Module `__getattr__` is only 3.7+, so without it Python 3.6 will raise `ImportError` when importing `PersonNameUnicode`

Reply:
But does it not mean that the deprecated alias itself won't be available in Python 3.6?

Reply:
> But does it not mean that the deprecated alias itself won't be available in Python 3.6?

I'm not sure what you mean. `PersonNameUnicode` is importable for 3.6+, with 3.7+ you get the deprecation warning. 

If lines 902-903 weren't there then it wouldn't be importable with 3.6. 

If line 902 wasn't there you wouldn't get a deprecation warning with 3.7+

Reply:
> I'm not sure what you mean. `PersonNameUnicode` is importable for 3.6+, with 3.7+ you get the deprecation warning.

Right,  sorry, I misread something.